From 4aa35f608e9a5d0c6ac2beaa7b10d090c6d10949 Mon Sep 17 00:00:00 2001
From: Christian Kampka <christian@kampka.net>
Date: Mon, 13 Aug 2018 21:09:01 +0200
Subject: [PATCH 2/2] Create paths if the don't exist

---
 cmd/root.go | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/cmd/root.go b/cmd/root.go
index fcfb30b..b53639b 100644
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -51,4 +51,18 @@ func initConf() {
 		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
 		os.Exit(1)
 	}
+
+	if histPath := filepath.Dir(config.Conf.History.Path.Abs()); histPath != "" {
+		if _, err := os.Stat(histPath); err != nil {
+			fmt.Printf("Creating directory '%s' for history storage", histPath)
+			os.MkdirAll(histPath, 0700)
+		}
+	}
+
+	if backupPath := config.Conf.History.BackupPath.Abs(); backupPath != "" {
+		if _, err := os.Stat(backupPath); err != nil {
+			fmt.Printf("Creating directory '%s' for backup storage", backupPath)
+			os.MkdirAll(backupPath, 0700)
+		}
+	}
 }
-- 
2.16.4

